### PR TITLE
feat(): added branching information

### PR DESCRIPTION
**Pull request by sandykumar93**
- added branching 101 section
- updated hyperlinks and readme

### CODE REVIEW

File: branching-101/branch-naming-convention.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+### Branch Naming Conventions 
+
+#### long running branches naming convention
+1. development
+2. testing / staging / uat
+3. production
+
+#### Feature branches naming convention
+{issue-type}/{jira/linear-ticket-number}-{short-description}
```
Comment (vishwasnavadak) [inline branching-101/branch-naming-convention.md:9]: Mention issue types and examples.

File: branching-101/branch-rules.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+## 🌿 Branch Protection Rules
+
+- Require a pull request before merging.
+    - Require approvals (at least 1) 👍.
+    - Require review from Code Owners (optional) 👨‍💻.
+- Require status checks to pass before merging 🛠️.
+    - Require branches to be up to date before merging (mention SonarCloud Analysis 📈, SNYK 🔒).
+- Require conversation resolution before merging 💬.
+- Require signed commits ✍️.
+- Do not allow bypassing the above settings ⛔.
```
Comment (vishwasnavadak) [inline branching-101/branch-rules.md:10]: We could attach screenshots, just to avoid confusion

**Review by vishwasnavadak — CHANGES_REQUESTED** [review]

File: branching-101/branch-naming-convention.md (hunk 1 above)
Comment (sandykumar93) [reply branching-101/branch-naming-convention.md:9]: done

File: branching-101/branch-rules.md (hunk 2 above)
Comment (sandykumar93) [reply branching-101/branch-rules.md:10]: done

**Review by vishwasnavadak — APPROVED** [review]

**Pull request merged**